upt_macports/upt_macports.py

```diff
@@ -51,6 +51,27 @@ def run_depends(self):
     def test_depends(self):
         return self._depends('test')
 
+    @property
+    def archive_type(self):
+        archive_keyword = {
+            'gz': 'gz',
+            '7z': '7z',
+            'bz2': 'bzip2',
+            'lzma': 'lzma',
+            'tar': 'tar',
+            'zip': 'zip',
+            'xz': 'xz'
+        }
+        try:
+            archive_name = self.upt_pkg.get_archive().filename
+            archive_type = archive_name.split('.')[-1]
+            return archive_keyword.get(archive_type, 'unknown')
+
+        except upt.ArchiveUnavailable:
+            self.logger.error(f'Detection of archive type failed: no archive \
+found')
```

Comment:
I am not a fan of this indentation, but I know you did that to get the text continued correctly. I would not be against just keeping this on one line and violating the "80-length" rule a bit, but perhaps the tests would complain about that....  Another option would be:
```
self.logger.error(f'Detection of archive type failed: no archive ' \
                   'found')
```

besides that (even with this) I would be happy to merge!

Reply:
Ok so I will keep it normal (violate 80-length) as that looks better and then maybe in other PR or if you are fine then in this same PR change the test to allow that rule violation as @mojca said.